client/network/src/protocol.rs

```diff
@@ -509,7 +509,6 @@ impl<B: BlockT> NetworkBehaviour for Protocol<B> {
 							) {
 								Ok(handshake) => {
 									let roles = handshake.roles;
-									self.peers.insert(peer_id, roles);
```

Comment:
This is removed because it is done as part of the while loop above?

Reply:
It was incorrectly there in the first place but `disconnect_peer()` would remove if the peer was rejected so it worked.